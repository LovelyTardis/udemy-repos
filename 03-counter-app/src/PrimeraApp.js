import PropTypes from 'prop-types';



const PrimeraApp = ({ saludo, subtitulo }) => {


    return (
        <>
            <h1>{saludo}!!!</h1>
            <p id='texto'>{subtitulo}</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subt√≠tulo'
}

export default PrimeraApp;