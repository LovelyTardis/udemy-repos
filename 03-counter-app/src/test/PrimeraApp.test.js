import React from "react";
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme'
import '@testing-library/jest-dom/extend-expect';

describe('Pruebas en <PrimeraApp />', () => {
    // test('Debe mostrar el mensaje "Hola, buenas"', () => {
    //     const saludo = 'Hola, buenas';

    //     const { getByText } = render( <PrimeraApp saludo={saludo} /> );

    //     expect( getByText(saludo) ).toBeInTheDocument();
    // });

    test('debe mostrar correctamente <PrimeraApp />', () => {
        const saludo = 'Hola, buenas';
        const wrapper = shallow( <PrimeraApp saludo={saludo}/> );

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar correctamente el subtítulo', () => {
        const saludo = 'Hola, buenas';
        const subTitulo = 'Soy un subtítulo';
        const wrapper = shallow( <PrimeraApp saludo={saludo} subtitulo={subTitulo}/> );

        const textoParrafo = wrapper.find('#texto').text();
        expect(textoParrafo).toBe(subTitulo);
    });
});