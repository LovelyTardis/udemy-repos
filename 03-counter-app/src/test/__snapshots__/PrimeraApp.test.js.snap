// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar correctamente <PrimeraApp /> 1`] = `
<Fragment>
  <h1>
    Hola, buenas
    !!!
  </h1>
  <p
    id="texto"
  >
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
